lib/stdlib/src/features.erl

```diff
@@ -0,0 +1,308 @@
+                false -> F(Word)
+            end
+    end.
+
+-spec resword_remove_feature(atom(), fun((atom()) -> boolean())) ->
+          {'ok', fun((atom()) -> boolean())}
+              | {'error', error()}.
+resword_remove_feature(Feature, F) ->
+    case is_valid_feature(Feature) of
+        true ->
+            {ok, remove_feature(Feature, F)};
+        false ->
+            {error, {?MODULE, {invalid_features, [Feature]}}}
+    end.
+
+-spec resword_add_features([atom()], fun((atom()) -> boolean())) ->
+          {'ok', fun((atom()) -> boolean())}
+              | {'error', error()}.
+resword_add_features(Features, F) ->
+    case lists:all(fun is_valid_feature/1, Features) of
+        true ->
+            {ok, lists:foldl(fun add_feature/2, F, Features)};
+        false ->
+            IsInvalid = fun(Ftr) -> not is_valid_feature(Ftr) end,
+            Invalid = lists:filter(IsInvalid, Features),
+            {error, {?MODULE, {invalid_features, Invalid}}}
+    end.
+
+-spec resword_remove_features([atom()], fun((atom()) -> boolean())) ->
+          {'ok', fun((atom()) -> boolean())}
+              | {'error', error()}.
+resword_remove_features(Features, F) ->
+    case lists:all(fun is_valid_feature/1, Features) of
+        true ->
+            {ok, lists:foldl(fun remove_feature/2, F, Features)};
+        false ->
+            IsInvalid = fun(Ftr) -> not is_valid_feature(Ftr) end,
+            Invalid = lists:filter(IsInvalid, Features),
+            {error, {?MODULE, {invalid_features, Invalid}}}
+    end.
+
+format_error(Reason, [{_M, _F, _Args, Info}| _St]) ->
+    ErrorInfo = proplists:get_value(error_info, Info, #{}),
+    ErrorMap = maps:get(cause, ErrorInfo),
+    ErrorMap#{reason => io_lib:format("~p: ~p", [?MODULE, Reason])}.
+
+format_error({invalid_features, Features}) ->
+    Fmt = fun F([Ftr]) -> io_lib:fwrite("'~p'", [Ftr]);
+              F([Ftr1, Ftr2]) ->
+                  io_lib:fwrite("'~p' and '~p'", [Ftr1, Ftr2]);
+              F([Ftr| Ftrs]) ->
+                  io_lib:fwrite("'~p', ~s", [Ftr, F(Ftrs)])
+          end,
+    case Features of
+        [Ftr] ->
+            io_lib:fwrite("the feature ~s does not exist.", [Fmt([Ftr])]);
+        Ftrs ->
+            io_lib:fwrite("the features ~s do not exist.", [Fmt(Ftrs)])
+    end.
+
+%% Hold the state of which features are currently enabled.
+%% This is almost static, so we go for an almost permanent state,
+%% i.e., user persistent_term.
+init_features() ->
+    persistent_term:put(enabled_features, []),
+    persistent_term:put(reserved_words, []),
+    case init:get_argument('enable-feature') of
+        error ->
+            %% no features enabled
+            ok;
+        {ok, Ftrs} ->
+            %% FIXME Need to be paranoid about any failures here.
+            %% Only convert to an existing atom.  This will also catch
+            %% a too long atom.
+            F = fun(String) ->
+                        case catch list_to_existing_atom(String) of
```

Comment:
I suggest using try/catch. We try to avoid the old-fashioned catch in new code (except test suites).